e2e_tests/tests/experiment/test_launch.py

```diff
@@ -0,0 +1,58 @@
+from typing import Callable
+
+import pytest
+
+from determined.experimental import Determined
+from tests import config as conf
+from tests import experiment as exp
+
+
+@pytest.mark.parallel
+def test_launch_layer_cifar(collect_trial_profiles: Callable[[int], None]) -> None:
+    config = conf.load_config(conf.cv_examples_path("cifar10_pytorch/const.yaml"))
+    config = conf.set_max_length(config, {"batches": 200})
+    config = conf.set_slots_per_trial(config, 1)
+    config = conf.set_profiling_enabled(config)
+    config = conf.set_entrypoint(
+        config, "python3 -m determined.launch.autohorovod model_def:CIFARTrial"
+    )
+
+    experiment_id = exp.run_basic_test_with_temp_config(
+        config, conf.cv_examples_path("cifar10_pytorch"), 1
+    )
+    trials = exp.experiment_trials(experiment_id)
+    (
+        Determined(conf.make_master_url())
+        .get_trial(trials[0]["id"])
+        .select_checkpoint(latest=True)
+        .load(map_location="cpu")
+    )
+
+    collect_trial_profiles(trials[0]["id"])
+
+    assert exp.check_if_string_present_in_trial_logs(
+        trials[0]["id"], "container exited successfully with a zero exit code"
+    )
+
+
+@pytest.mark.parallel
+def test_launch_layer_exit(collect_trial_profiles: Callable[[int], None]) -> None:
+    config = conf.load_config(conf.cv_examples_path("cifar10_pytorch/const.yaml"))
+    config = conf.set_max_length(config, {"batches": 200})
+    config = conf.set_slots_per_trial(config, 1)
+    config = conf.set_profiling_enabled(config)
+    config = conf.set_entrypoint(
+        config, "python3 -m nonexistent_launch_module model_def:CIFARTrial"
+    )
+
+    experiment_id = exp.run_failure_test_with_temp_config(
+        config, conf.cv_examples_path("cifar10_pytorch")
+    )
+    trials = exp.experiment_trials(experiment_id)
+    (Determined(conf.make_master_url()).get_trial(trials[0]["id"]))
+
+    collect_trial_profiles(trials[0]["id"])
+
+    assert exp.check_if_string_present_in_trial_logs(
+        trials[0]["id"], "container failed with non-zero exit code: 1"
+    )
```

Comment:
is this going to annoyingly say "assertion failed: False" if it fails?

Reply:
nope, because i added a print statement inside of check_if_string_present_in_trial_logs a while back that dumps the trial logs and the string that was not found.

Reply:
ah.  Someday we should maybe make it `exp.assert_string_present_in_trial_logs()`.  But no need to do that today.